### ISSUE TITLE
Add an unique id for every entry in history

### ISSUE THREAD

**Issue by Anindyadeep** [stanfordnlp/dspy#1073]
We can infer that, history in dspy acts as a trace, where we can see previous LLM response along with all the additional arguments. However, with more and more llm responses, it becomes hard to follow along the traces, since this is a list. And this becomes a common problem for all the supported LLMs in `dsp.modules` 

So my proposal is to have some id parameter with the following example standard:

```
module-name:current-date-time
```

or we can also use uuid but then add, date-time parameter seperately. The reason I put emphasis on datetime, because it makes easier to do some filter operations. 

Let me know what everyone thinks on this.

**Comment by tom-doerr** [comment]
Using date-time might also help in time-performance analysis

**Comment by Anindyadeep** [comment]
Yes, I agree

**Comment by okhat** [comment]
This is solid idea. @Anindyadeep wanna add a PR to the new dspy.LM client?

**Comment by Anindyadeep** [comment]
> This is solid idea. @Anindyadeep wanna add a PR to the new dspy.LM client?

sure, would be happy to do it

**Comment by Anindyadeep** [comment]
> This is solid idea. @Anindyadeep wanna add a PR to the new dspy.LM client?

I have one question though should it be added on the `dspy/clients.lm` or `dspy/modules.lm` or both? Since module seems to be the base class?

**Comment by Anindyadeep** [comment]
Hey, also I added an another issue, while I was solving this one, let me know your thoughts.
